### PR TITLE
[improve] Upgrade httpclient version to 5.1

### PR DESCRIPTION
**Pull request by wolfboys**
[improve] Upgrade httpclient version to 5.1

### CODE REVIEW

File: streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala [hunk 1]
```diff
@@ -16,23 +16,23 @@
...
-import org.apache.http.client.methods._
-import org.apache.http.client.utils.URIBuilder
-import org.apache.http.config.RegistryBuilder
-import org.apache.http.entity.StringEntity
-import org.apache.http.impl.auth.SPNegoSchemeFactory
-import org.apache.http.impl.client.{BasicCredentialsProvider, CloseableHttpClient, HttpClientBuilder, HttpClients}
-import org.apache.http.impl.conn.PoolingHttpClientConnectionManager
-import org.apache.http.message.BasicNameValuePair
-import org.apache.http.util.EntityUtils
+import org.apache.hadoop.shaded.org.apache.http.client.config.AuthSchemes
```
Comment (zhoulii) [inline streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala]: We'd better not use reference from `hadoop.shaded` package.

File: streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala [hunk 2]
```diff
@@ -97,83 +98,47 @@ object HttpClientUtils {
...
+        def getUserPrincipal: Principal = null
+      }
+
+      credentialsProvider.setCredentials(new AuthScope(null, -1), credentials)
 
       val authSchemeRegistry = RegistryBuilder
-        .create[AuthSchemeProvider]
-        .register(AuthSchemes.SPNEGO, new SPNegoSchemeFactory(true))
+        .create[AuthSchemeFactory]
+        .register(AuthSchemes.SPNEGO, SPNegoSchemeFactory.DEFAULT)
```
Comment (zhoulii) [inline streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala]: `AuthSchemes` is removed in 5.1, we can use `StandardAuthScheme` instead.

**Review by zhoulii — COMMENTED** [review]
Nice work, I left a minor comment, please take a look.

File: streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala (hunk 2 above)
Comment (wolfboys) [reply streampark-common/src/main/scala/org/apache/streampark/common/util/HttpClientUtils.scala]: > `AuthSchemes` is removed in 5.1, we can use `StandardAuthScheme` instead.

Thanks for your review, I'll update it later

**Comment by zhoulii** [comment]
Deployed and tested with yarn `application/session/perjob` execution mode, no bug was found, so LGTM.

**Comment by wolfboys** [comment]
> Deployed and tested with yarn `application/session/perjob` execution mode, no bug was found, so LGTM.

thanks for your review and test.

**Review by wangsizhu0504 — APPROVED** [review]
LGTM

**Pull request merged**